openmedia/gui/frames.py

```diff
@@ -0,0 +1,193 @@
+            button.bind(button.sequence, getattr(self, '_' + button.name + '_handler'))
+        for button in self.f_playlist.buttons:
+            button.bind(button.sequence, getattr(self, '_' + button.name + '_handler'))
+        self.f_control.s_volume.bind('<B1-Motion>', self._set_volume)
+        self.f_progress.s_progress.bind('<ButtonRelease-1>', self._skip)
+        self.f_playlist.l_playlist.bind('<Double-Button-1>', self._play)
+        self.f_progress.set_alarm(self._progress)
+
+    def _update_play_icon(self, button):
+        if mixer.is_playing():
+            button.config(text=u'▶')
+        else:
+            button.config(text=u'▌▌')
+
+    def _play_pause(self, button):
+        if mixer.is_playing():
+            button.config(text=u'▶')
+            mixer.pause()
+        else:
+            button.config(text=u'▌▌')
+            mixer.play()
+
+    def _play_pause_handler(self, event):
+        self._play_pause(event.widget)
+
+    def _stop_handler(self, event):
+        self._play_pause(self.f_control.buttons[_ControlFrame.PLAY])
+        mixer.stop()
+        self.f_progress.set(0)
+
+    def _play_next(self, button):
+        if mixer.is_paused:
+            self._play_pause(self.f_control.buttons[_ControlFrame.PLAY])
+        list_box = self.f_playlist.l_playlist
+        list_box.highlight_next(mixer.song_index)
+        self.f_progress.set(0)
+        mixer.play_next()
+        self.f_progress.set_max(mixer.get_song_duration())
+
+    def _play_next_handler(self, event):
+        self._play_next(event.widget)
+
+    def _show_hide_handler(self, event):
+        if event.widget.checked is None:
+            raise InvalidWidgetStateException('Expected boolean value for ' +
+                                              str(event.widget))
+        if event.widget.checked:
+            self.f_playlist.l_playlist.grid_remove()
+            self.f_playlist.yscroll.grid_remove()
+            self.f_playlist.buttons[_PlaylistFrame.ADD].grid_remove()
+            event.widget.checked = False
+        else:
+            self.f_playlist.l_playlist.grid(row=0, column=0, columnspan=2)
+            self.f_playlist.yscroll.grid(row=0, column=2, sticky=Tk.N+Tk.S)
+            self.f_playlist.buttons[_PlaylistFrame.ADD].grid(row=1, column=1)
+            event.widget.checked = True
+
+    def _play(self, event):
+        mixer.stop()
+        mixer.play(self.f_playlist.l_playlist.get(Tk.ACTIVE))
+        self.f_control.buttons[_ControlFrame.PLAY].config(text=u'▌▌')
+        progress = self.f_progress
+        progress.set(0)
+        progress.set_max(mixer.get_song_duration())
+
+    def _add_handler(self, event):
+        self.filename = os.path.basename(tkFileDialog.askopenfilename())
+        self.f_playlist.l_playlist.add_track(self.filename)
+        mixer.add(self.filename)
+
+    def _set_volume(self, event):
+        mixer.set_volume(float(event.widget.get()) / 100)
+
+    def _progress(self):
+        bar = self.f_progress
```

Comment:
`self.f_progress` is not an actually progress bar. Let's rename it or use `self.f_progress` instead.